### PR TITLE
(PC-12676) feat(Form): added Web form and upgrade react-native, reactnative-web and @types/react-native

### PR DESCRIPTION
**Pull request by kopax-polyconseil**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-12676

## Checklist

I have:

- [ ] Made sure the title of my PR follows the [convention](1) `($jira) $type($scope): $summary`.
- [ ] Made sure my feature is working on the relevant real / virtual devices (native and web).
- [ ] Written **unit tests** native (and web when implementation is different) for my feature.
- [ ] Added a **screenshot** for UI tickets.
- [ ] Attached a **ticket number** or **github dev name** for any added TODO / FIXME.
- [ ] Added new reusable components to **AppComponents** page and communicate to the team on slack.
- [ ] If my PR is a bugfix, I add the link of the "résolution de problème sur le bug" [on Notion](2)
- [ ] Made sure translations file is up to date (`yarn translations:extract`)

## Screenshots

| iOS | Android | Mobile - Chrome | Desktop - Chrome | Desktop - Safari |
| --: | ------: | --------------: | ---------------: | ---------------: |
|     |         |                 |                  |                  |

[1]: https://github.com/pass-culture/pass-culture-app-native/blob/master/doc/standards/pr-title.md
[2]: https://www.notion.so/passcultureapp/R-solution-de-probl-mes-sur-les-bugs-5dd6df8f6a754e6887066cf613467d0a

### CODE REVIEW

File: src/ui/components/OrSeparator.tsx [hunk 1]
```diff
@@ -11,6 +11,7 @@ interface OrSeparatorProps extends ViewProps {
 
 export const OrSeparator: React.FC<OrSeparatorProps> = ({ lineColor, ...props }) => {
   return (
+    // @ts-ignore TODO: fix when https://github.com/necolas/react-native-web/issues/2189#issuecomment-1008886405 is resolved
```
Comment (Lucasbeneston) [inline src/ui/components/OrSeparator.tsx:14]: Je ne comprends pas le lien entre ce TODO et le composant "OrSperator" ?

File: src/features/search/pages/Search.tsx [hunk 2]
```diff
@@ -54,8 +55,10 @@ export function Search() {
 
   return (
     <Container>
-      <SearchHeader />
-      {showResults ? <SearchResults /> : <SearchLandingPage />}
+      <Form>
+        <SearchHeader />
+        {showResults ? <SearchResults /> : <SearchLandingPage />}
+      </Form>
```
Comment (Lucasbeneston) [inline src/features/search/pages/Search.tsx:61]: 👍

**Review by Lucasbeneston — APPROVED** [review]
LGTM 👍

File: src/ui/components/OrSeparator.tsx (hunk 1 above)
Comment (kopax-polyconseil) [reply src/ui/components/OrSeparator.tsx:14]: Bonne remarque, la maj de react native à remonté une erreur qui concerne `accessibilityRole`, qui n'est toujours pas solutionné, donc j'ai copié le commentaire.

Comment (Lucasbeneston) [reply src/ui/components/OrSeparator.tsx:14]: Du coup on peut supprime ce commentaire ? Car il n'y a pas d' `accessibilityRole` sur ce composant.

Comment (kopax-polyconseil) [reply src/ui/components/OrSeparator.tsx]: MAJ

**Pull request merged**